tonic/src/transport/endpoint.rs

```diff
@@ -259,18 +265,47 @@ impl ClientTlsConfig {
     }
 
     /// Sets the domain name against which to verify the server's TLS certificate.
+    ///
+    /// This has no effect if `rustls_client_config` or `openssl_connector` is used to configure
```

Comment:
so I wonder if there is a way we can make this actually work? From what I can tell everything just takes a &mut self so you can in theory use the passed config and just apply these changes to it. If they don't pass a config then its just the base version that we apply changes too?

Reply:
As discussed I've opened #116 to discuss this further since it's likely a separate change.

Adding these comments documents the current behaviour, which in the first design iteration of this PR was important. We could remove the docs now we don't add any new builder methods, but I don't think the extra clarity harms anything.